syft/pkg/cataloger/cpe_candidate.go

```diff
@@ -0,0 +1,107 @@
+package cataloger
+
+import (
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/scylladb/go-set/strset"
+)
+
+// this is a static mapping of known package names (keys) to official cpe names for each package
+type candidatesByPackageType map[pkg.Type]map[string][]string
+
+type filterCpeFieldCandidateFn func(cpeFieldCandidate) bool
+
+type cpeFieldCandidate struct {
+	value                       string
+	disallowSubSelections       bool
+	disallowDelimiterVariations bool
+}
+
+type cpeFieldCandidateSet struct {
```

Comment:
this should probably be `type cpeFieldCandidateSet map[cpeFieldCandidate]struct{}`

Reply:
updated in 2b37399218d52fc81d067179213ce1c52616f33a